### ISSUE TITLE
Reproducing CoCLR RGB Results???

### ISSUE THREAD

**Issue by srijandas07** [TengdaHan/CoCLR#43]
I have tested the CoCLR RGB model provided in the repo. The results I get is actually better than the ones provided in the paper. However, when I try reproducing the CoCLR RGB model by training from scratch, the improvement is something around 10 % compared to InfoNCE model. I did not manage to reproduce the jump in accuracy from 45 % to 70% for the RGB model with one cycle (FlowMining). 
Btw, while performing late fusion with RGB and Flow - the results again matches with the ones provided in the paper (due to their complementary nature).
Is there something I am missing while training CoCLR-RGB?

**Comment by TengdaHan** [comment]
Hi, thanks for pointing out the issue! I found the problem:

**Problem:** the current training loss has lower efficiency especially for small pre-train datasets (like UCF101), to reproduce the same results it needs a longer training schedule (about 300 epochs) rather than 100 epochs.

**Reason:** in the CoCLR training, the model has been trained with InfoNCE loss, so the self-similarity is already very high, resulting in the MIL-NCE loss being very low at the beginning. So the Co-training is not efficient because the gradient is already very small.

**Fix:** we randomly mask out the self-similarity in the MIL-NCE loss. I have updated the GitHub code, you can check https://github.com/TengdaHan/CoCLR/blob/main/main_coclr.py#L382. In our old experiments, this "randomly-masking" operation has no effect on K400 CoCLR training because in the previous InfoNCE training stage the self-similarity is not very low at the end of the training schedule. But the "randomly-masking" operation will accelerate CoCLR training on smaller datasets like UCF101. Since our main experiments were on Kinetics, we didn't report this trick (specifically for small datasets) in our paper. If you want to reproduce the results on UCF101, you can have a try with this randomly-masking turned on. 

**History:** Actually I had this part of the code in my repo at the beginning. But it was deleted somehow when I cleaned my code. It is my bad. See history below:
https://github.com/TengdaHan/CoCLR/commit/c95eba977b68f713dca51c8c43c4fb64d69bcf59#diff-fbd86e95c658177ef10b81dcd01ddb50d93152bfc5d109981994436a834a0727L376

Anyway, thank you very much for notifying me of this issue.
